react/components/SearchBar/SearchBar.tsx

```diff
@@ -159,11 +208,13 @@ const SearchBar = ({
 
   const SelectedAutocompleteResults = useMemo(() => {
     return isAutocompleteDeclared
-      ? props => <ExtensionPoint id={AUTCOMPLETE_EXTENSION_ID} {...props} />
-      : props => <AutocompleteResults {...props} />
+      ? (props: AutocompleteResultsProps) => (
+          <ExtensionPoint id={AUTCOMPLETE_EXTENSION_ID} {...props} />
+        )
+      : (props: AutocompleteResultsProps) => <AutocompleteResults {...props} />
   }, [isAutocompleteDeclared])
```

Comment:
Not related to the migration, but this could be refactored to if() {return ... } return ...